rest/admin-api/src/main/java/org/keycloak/admin/api/client/ClientApi.java

```diff
@@ -1,14 +1,31 @@
 package org.keycloak.admin.api.client;
 
+import jakarta.ws.rs.Consumes;
 import jakarta.ws.rs.GET;
+import jakarta.ws.rs.PUT;
+import jakarta.ws.rs.PathParam;
 import jakarta.ws.rs.Produces;
-import jakarta.ws.rs.QueryParam;
 import jakarta.ws.rs.core.MediaType;
+
+import org.keycloak.admin.api.FieldValidation;
 import org.keycloak.representations.admin.v2.ClientRepresentation;
 
+import com.fasterxml.jackson.databind.JsonNode;
+
 public interface ClientApi {
 
     @GET
     @Produces(MediaType.APPLICATION_JSON)
     ClientRepresentation getClient();
+
+    @PUT
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.APPLICATION_JSON)
+    ClientRepresentation createOrUpdateClient(ClientRepresentation client, @PathParam("fieldValidation") FieldValidation fieldValidation);
+
+    @PUT
+    @Consumes("application/merge-patch+json")
+    @Produces(MediaType.APPLICATION_JSON)
+    ClientRepresentation patchClient(JsonNode patch, @PathParam("fieldValidation") FieldValidation fieldValidation);
```

Comment:
Field validation might need a better name not to be confused with regular validations (from semantic perspective etc.). Possibly something like `unknownFieldsPolicy` but not sure.

Also should we really give the power to set this policy to the user? Shouldn't it rather be a global server option? In other words, do we really want to set this per request?

Reply:
> Also should we really give the power to set this policy to the user? Shouldn't it rather be a global server option? In other words, do we really want to set this per request?

You would probably want control per request yes. You can certainly make a case for session or server level control as well.